tensorflow/core/common_runtime/dml/dml_buffer_region.cc

```diff
@@ -20,23 +20,35 @@ limitations under the License.
 namespace tensorflow {
 
 D3D12BufferRegion::D3D12BufferRegion(
-    D3D12HeapAllocator* allocator, uint32_t allocation_id,
-    Microsoft::WRL::ComPtr<ID3D12Resource> resource, uint64_t offset,
-    uint64_t size_in_bytes)
-    : allocator_(allocator),
-      allocation_id_(allocation_id),
+    uint64_t offset, uint64_t size_in_bytes,
+    D3D12_RESOURCE_STATES resource_state,
+    Microsoft::WRL::ComPtr<ID3D12Resource> resource,
+    Microsoft::WRL::ComPtr<ID3D12Resource> resource_copy_src_state,
+    Microsoft::WRL::ComPtr<ID3D12Resource> resource_copy_dst_state)
+    : offset_(offset),
+      size_in_bytes_(size_in_bytes),
+      resource_state_(resource_state),
       resource_(std::move(resource)),
-      offset_(offset),
-      size_in_bytes_(size_in_bytes) {
+      resource_copy_src_state_(std::move(resource_copy_src_state)),
+      resource_copy_dst_state_(std::move(resource_copy_dst_state)) {
   assert(resource_ != nullptr);
   assert(size_in_bytes_ != 0);
   assert(resource_->GetDesc().Dimension == D3D12_RESOURCE_DIMENSION_BUFFER);
+  // TODO: if other resources provided they should match size
 }
 
-ID3D12Resource* D3D12BufferRegion::Resource() const {
+ID3D12Resource* D3D12BufferRegion::ResourceInFixedState() const {
   return resource_ ? resource_.Get() : nullptr;
 }
 
+ID3D12Resource* D3D12BufferRegion::ResourceInCopySrcState() const {
+  return resource_copy_src_state_ ? resource_copy_src_state_.Get() : nullptr;
```

Comment:
[minor] Is not equivalent to just say `return resource_copy_src_state_.Get()`? If it's false, then the ComPtr would be nullptr anyway.

Reply:
Makes sense. Just copying what was here before. :)